content/reader_test.go

```diff
@@ -0,0 +1,225 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package content
+
+import (
+	"bytes"
+	_ "crypto/sha256"
+	"errors"
+	"io"
+	"testing"
+
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+func TestVerifyReader_Read(t *testing.T) {
+	// matched content and descriptor with small buffer
+	content := []byte("example content")
+	desc := ocispec.Descriptor{
+		MediaType: ocispec.MediaTypeImageLayer,
+		Digest:    digest.FromBytes(content),
+		Size:      int64(len(content))}
+	r := bytes.NewReader(content)
+	vr := NewVerifyReader(r, desc)
+	buf := make([]byte, 5)
+	n, err := vr.Read(buf)
+
+	if err != nil {
+		t.Fatal("Read() error = ", err)
+	}
+	if !bytes.Equal(buf, []byte("examp")) {
+		t.Fatalf("incorrect read content: %s", buf)
+	}
+	if n != 5 {
+		t.Fatalf("incorrect number of bytes read: %d", n)
+	}
+
+	// matched content and descriptor with sufficient buffer
+	content = []byte("foo foo")
+	desc = ocispec.Descriptor{
+		MediaType: ocispec.MediaTypeImageLayer,
+		Digest:    digest.FromBytes(content),
+		Size:      int64(len(content))}
+	r = bytes.NewReader(content)
+	vr = NewVerifyReader(r, desc)
+	buf = make([]byte, len(content))
+	n, err = vr.Read(buf)
+	if err != nil {
+		t.Fatal("Read() error = ", err)
+	}
+	if n != len(content) {
```

Comment:
Do we want to compare the content also?

Reply:
Yes I think we do. Added content check here.